hypertrace-core-graphql-attribute-store/src/main/java/org/hypertrace/core/graphql/attributes/CachingAttributeStore.java

```diff
@@ -54,8 +56,19 @@
 
   @Override
   public Single<List<AttributeModel>> getAll(GraphQlRequestContext requestContext) {
-    return GrpcRxExecutionContext.forContext(this.grpcContextBuilder.build(requestContext))
-        .wrapSingle(this.cachingAttributeClient::getAll)
+    List<AttributeMetadata> unfilteredList =
+        GrpcRxExecutionContext.forContext(this.grpcContextBuilder.build(requestContext))
+            .wrapSingle(this.cachingAttributeClient::getAll)
+            .blockingGet();
```

Comment:
No blocking! defeats the purpose of working in async. 

Filtering here is actually trivial, right before translating on line 73, you could add a filter function like
```java
        .filter(attribute -> !attribute.getInternal())
```

But better yet, what we really should be doing is changing our request to filter these out. The caching attribute client accepts a default filter to use on all requests, so we can set it to only request external attributes. If that attribute isn't exposed on the filter, we should add it.

Reply:
It looks like it wasn't added there - https://github.com/hypertrace/attribute-service/blob/main/attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto#L134-L139 - so let's go ahead and add support there first, since filtering here should be a pretty typical case.

Reply:
> If that attribute isn't exposed on the filter, we should add it.

Did you mean to add support for internal attribute filtering in `AttributeMetadataFilter`? https://github.com/hypertrace/attribute-service/blob/2be10b06b1309712c11e365eda6915378f0bd4b9/attribute-service-api/src/main/proto/org/hypertrace/core/attribute/service/v1/attribute_metadata.proto#L134

Reply:
My bad. I had an old state of the PR open. Just looked at your latest comment